### PR TITLE
Implemented: Feature to run scheduled job right away(#364ttgf)

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

Closes #

### Screenshots of Visual Changes before/after (If There Are Any)

https://user-images.githubusercontent.com/62360781/183361711-8cf25b54-0730-445d-89e1-4b595114a6d1.mov

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/components/JobActionsPopover.vue [hunk 1]
```diff
@@ -89,17 +93,31 @@ export default defineComponent({
...
     return {
       copyOutline, 
+      flashOutline,
       pinOutline,
       store, 
       timeOutline  
     }
   }
 });
-</script>
```
Comment (ymaheshwari1) [inline src/components/JobActionsPopover.vue:105]: Unwanted change.

File: src/components/JobActionsPopover.vue [hunk 2]
```diff
@@ -89,17 +93,31 @@ export default defineComponent({
         await this.store.dispatch('user/updatePinnedJobs', { pinnedJobs: [...pinnedJobs] });
       }
       this.closePopover();
+    },
+    async runJobNow(job: any) {
+      if(job) {
+        const { Clipboard } = Plugins;
+        const jobDetails = `jobId: ${job.jobId}, jobName: ${this.getEnumName(job.systemJobEnumId)}, jobDescription: ${this.getEnumDescription(job.systemJobEnumId)}`;
+
+        await Clipboard.write({
```
Comment (ymaheshwari1) [inline src/components/JobActionsPopover.vue:102]: Why we have added logic to copy the job details on clicking run now?

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/components/JobActionsPopover.vue (hunk 2 above)
Comment (shashwatbangar) [reply src/components/JobActionsPopover.vue:102]: Sir it was mentioned in the ticket itself.
Please have a look - https://app.clickup.com/t/364ttgf

File: src/components/JobActionsPopover.vue (hunk 1 above)
Comment (shashwatbangar) [reply src/components/JobActionsPopover.vue:105]: Reverting it.

File: src/components/JobActionsPopover.vue (hunk 2 above)
Comment (ymaheshwari1) [reply src/components/JobActionsPopover.vue:102]: Ok, it means that we need to `copy the job`(use the same job for run now) and just update the recurrence count, we don't need to copy job details.

File: src/components/JobActionsPopover.vue [hunk 3]
```diff
@@ -89,13 +93,20 @@ export default defineComponent({
         await this.store.dispatch('user/updatePinnedJobs', { pinnedJobs: [...pinnedJobs] });
       }
       this.closePopover();
+    },
+    async runJobNow(job: any) {
+      if(job) {
+        await this.store.dispatch('job/runServiceNow', job)
```
Comment (adityasharma7) [inline src/components/JobActionsPopover.vue:99]: I think we should handle setting productStoreId from job instead of state value

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/JobActionsPopover.vue (hunk 3 above)
Comment (shashwatbangar) [reply src/components/JobActionsPopover.vue:99]: Yes sir. Understood as we discussed. Fixing it

File: src/store/modules/job/actions.ts [hunk 4]
```diff
@@ -525,7 +525,7 @@ const actions: ActionTree<JobState, RootState> = {
       'SERVICE_COUNT': '0',
       'SERVICE_TEMP_EXPR': job.jobStatus,
       'jobFields': {
-        'productStoreId': this.state.user.currentEComStore.productStoreId,
+        'productStoreId': job.status === "SERVICE_PENDING" ? job.productStoreId : this.state.user.currentEComStore.productStoreId,
```
Comment (ymaheshwari1) [inline src/store/modules/job/actions.ts:528]: Add the check: https://github.com/hotwax/job-manager/pull/211/files#diff-1edc9153c69edc14cbca5528effde857efdaf6da5ac6048e4d6fc896b11ffcb4R540

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/actions.ts (hunk 4 above)
Comment (shashwatbangar) [reply src/store/modules/job/actions.ts:528]: Yes sir. Understood why we need this check. Fixed it

File: src/components/JobActionsPopover.vue (hunk 3 above)
Comment (adityasharma7) [reply src/components/JobActionsPopover.vue:99]: Check if we need to handle for shopifyConfigId as well

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/JobActionsPopover.vue (hunk 3 above)
Comment (shashwatbangar) [reply src/components/JobActionsPopover.vue:99]: Sir shopifyConfigId never comes from job, thus I guess it is okay to always set it from userState

**Comment by dt2patel** [comment]
We should also add a confirmation alert that asks the user if they are sure they want run now.
Alert has been added in Figma.

**Comment by shashwatbangar** [comment]
> We should also add a confirmation alert that asks the user if they are sure they want run now.
> Alert has been added in Figma.

Done sir. Added an alert to confirm to run job immediately.

File: src/store/modules/job/actions.ts [hunk 5]
```diff
@@ -525,19 +525,19 @@ const actions: ActionTree<JobState, RootState> = {
...
         'systemJobEnumId': job.systemJobEnumId,
         'tempExprId': job.jobStatus, // Need to remove this as we are passing frequency in SERVICE_TEMP_EXPR, currently kept it for backward compatibility
         'parentJobId': job.parentJobId,
         'recurrenceTimeZone': this.state.user.current.userTimeZone
       },
-      'shopifyConfigId': this.state.user.shopifyConfigId,
+      'shopifyConfigId': job.shopifyConfigId ? job.shopifyConfigId : this.state.user.shopifyConfigId,
```
Comment (adityasharma7) [inline src/store/modules/job/actions.ts:534]:
```suggestion
      'shopifyConfigId': job.status === "SERVICE_PENDING" ? job.shopifyConfigId : this.state.user.shopifyConfigId,
```

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/JobActionsPopover.vue [hunk 6]
```diff
@@ -29,8 +33,9 @@ import {
   popoverController
 } from "@ionic/vue";
 import { defineComponent } from "vue";
-import { copyOutline, pinOutline, timeOutline  } from 'ionicons/icons'
+import { copyOutline, flashOutline, pinOutline, timeOutline  } from 'ionicons/icons'
 import { mapGetters, useStore } from 'vuex'
+import { alertController } from '@ionic/vue';
```
Comment (ymaheshwari1) [inline src/components/JobActionsPopover.vue:38]: Import ionic components in a single statement.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/actions.ts (hunk 5 above)
Comment (shashwatbangar) [reply src/store/modules/job/actions.ts:534]: Ok sir. Fixed it.

File: src/components/JobActionsPopover.vue (hunk 6 above)
Comment (shashwatbangar) [reply src/components/JobActionsPopover.vue:38]: Done sir.

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**